### PR TITLE
docs/examples: thorough overhaul

### PR DESCRIPTION
**Pull request by exo-cedric**
Updated all examples and documentation to reflect the current state of recommended resources (as opposed to _deprecated_ ones).

All examples are now:
- **self-contained**: each example is a fully working Terraform configuration, which may be `terraform apply`-ed without preliminary setup/dependencies
- contained in a single `main.tf` file (except for the recurring and "off-topic" `providers.tf` and `ssh.tf`), making it easier to handle and grasp
- consolidated to use the same naming convention (throughout all examples)
- linted, validated and **tested** using personal credentials:
```
terraform fmt
terraform init
terraform validate
terraform apply
```
- sporting a `README.md` file (with no exception)

Examples that overlapped have been removed/merged:
- `s3` (renamed) -> `sos`
- `cloud-init` (removed) -> `multipart-cloud-init`
- `multi-private-network` (merged) -> `managed-private-network`

### CODE REVIEW

**Comment by PhilippeChepy** [comment]
This PR is related to #187 (outdated documentation issue)

File: docs/guides/migration-of-security-group-rules.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ description: |-
 
 # Migrating from security_group_rules to security_group
 
+**This migration guide applies to Exoscale Terraform Provider version 0.31.3 or above.**
```
Comment (kobajagi) [inline docs/guides/migration-of-security-group-rules.md]: Terraform docs support card boxes (callouts):
https://www.terraform.io/registry/providers/docs#callouts

Not insisting it should be used here, just pointing out the fact.

File: examples/dns/README.md [hunk 2]
```diff
@@ -1,19 +1,48 @@
...
+my_instance_ipv4 = "185.19.31.186"
+my_instance_ipv6 = "2a04:c43:e00:63a3:43b:eaff:fe00:3f6"
 ```
 
-Then, on the Portal, you should see the following entries.
+Which can then be seen using [Exoscale CLI](https://github.com/exoscale/cli/):
 
-![expected output in the portal](example-exo.png)
+```console
+$ exo dns show example.exo
```
Comment (kobajagi) [inline examples/dns/README.md:31]: CLI output is not up-to-date.  Latest CLI (1.58.0) uses API v2 for DNS commands and IDs are now in UUID format.

**Comment by kobajagi** [comment]
Awesome work @exo-cedric !

**Comment by exo-cedric** [comment]
Thanks!

I'm putting that PR back in draft/WIP as I realize I ought to go through docs/resources/... (and maybe others docs/...) for similar consolidation

**Comment by exo-cedric** [comment]
`docs/resources` and `docs/data-sources` have been likewised overhauled.

**DEPRECATED** resources/data-sources:
- have all been duly verified/identified as such
- all point to the replacing sibling
- have been stripped from their Usage and Import section

Generally:
- _external_ links have been reduced to their bare minimum (in favor of internal links to appropriate resource/data-source)
- naming and wording have been unified/consolidated (including leaner wording whenever possible)
- Usage section has been reduced to demonstrating the resource/data-source (required) syntax (in favor of fully-fledge, self-contained, working examples in the ~/examples/ directory)

**Review by PhilippeChepy — APPROVED** [review]

File: examples/ipv6/README.md [hunk 3]
```diff
@@ -1,46 +1,36 @@
...
-This example show how to activate the IPv6 address feature `ip6 = true` as well as setting the Security Group Ingress rules to enable SSH ports.
+This example demonstrates how to activate
+[IPv6](https://community.exoscale.com/documentation/compute/ipv6/)
+on your compute instances, thanks to the `ipv6 = true` argument.
 
-Initializing Terraform will ask for the Exoscale provider and some credentials.
+Please refer to the [main.tf](./main.tf) Terraform configuration file.
 
-```
+One should node the IPv6-specific security groups rules set up for the SSH access
```
Comment (kobajagi) [inline examples/ipv6/README.md]: `s/node/note`

**Review by kobajagi — APPROVED** [review]

**Pull request merged**